src/translate.rs

```diff
@@ -1162,38 +1176,67 @@ impl FromGlib<i32> for Option<u64> {
     }
 }
 
-/// Translate from a pointer type without taking ownership, transfer: none.
+/// Translate from a pointer type which is annotated with `transfer: none`.
+/// The resulting value is referenced at least once, by the bindings.
+/// Values obtained using this trait must be unreferenced on `drop()`
+/// by the implementing type.
+///
+/// This is suitable for floating references, which become strong references.
+///
+/// For more information, refer to module level documentation.
 pub trait FromGlibPtrNone<P: Ptr>: Sized {
     unsafe fn from_glib_none(ptr: P) -> Self;
 }
 
-/// Translate from a pointer type taking ownership, transfer: full.
+/// Translate from a pointer type which is annotated with `transfer full`.
+/// This transfers the ownership of the value to the Rust side.
+/// Values obtained using this trait must be unreferenced on `drop()`
+/// by the implementing type.
+///
+/// Because ownership can only be transferred if something is already referenced,
+/// this is unsuitable for floating references.
+///
+/// For more information, refer to module level documentation.
 pub trait FromGlibPtrFull<P: Ptr>: Sized {
     unsafe fn from_glib_full(ptr: P) -> Self;
 }
 
-/// Translate from a pointer type by borrowing. Don't increase the refcount
+/// Translate from a pointer type by borrowing, without affecting the refcount.
+/// Values obtained using this trait must *not* be unreferenced on `drop()`
+/// by the implementing type.
+///
+/// It is not recommended to use this on floating references,
+/// which may get transferred to a callee and dropped,
+/// leaving the pointer stale.
```

Comment:
Actually because of one of the many special cases it is required to use this for floating references instead of `from_glib_none()`. Generally this one is only used in callbacks from C (signal handlers, virtual methods, proper callbacks) for getting access on the Rust side to a value that we only "borrow" during the scope of the callback. (see also what I wrote above about some rules for safely using it)

In case of floating references this does not sink them or anything but just keeps them as is. This is especially important for virtual method trampolines: otherwise when e.g. implementing a `GtkWidget` in Rust and overriding the `GObject::constructed` virtual method what would happen is that we would pass it to `from_glib_none()` on the Rust side. That would sink and take ownership of that floating reference then, and at the end of the virtual method we would drop it. Which then means that the caller (who just created a new instance!) would get an already freed pointer back from the constructor.

Now this is all detailed background information and should probably not go verbatim into the docs here :)